administrator/com_joomgallery/src/Service/Metadata/MetadataPHP.php

```diff
@@ -0,0 +1,600 @@
+   * 
+   * @param   string $img   Path to the image 
+   * @param   mixed  $edits Array of edits to be made to the metadata
+   * 
+   * @return  bool          True on success, false on failure
+   * 
+   * @since   4.0.0
+   */
+  public function writeToIptc(string $img, $edits): bool
+  {
+    $editor = new IptcDataEditor();
+    $tagString = "";
+    foreach ($edits as $tag => $edit) {
+      if ($edit != "") {
+        $result = $editor->createEdit($tag, $edit);
+        if ($result != false) {
+          $tagString .= $result;
+        }
+      }
+    }
+
+    // If no edits were made, then don't try to embed data.
+    if ($tagString == "") {
+      return true;
+    }
+
+    $content = iptcembed($tagString, $img);
+    $fp = fopen($img, "wb");
+    fwrite($fp, $content);
+    fclose($fp);
+    return true;
+  }
+
+  /**
+   * Reads the metadata from an image.
+   * (Current supported image formats: JPEG)
+   * (Current supported metadata formats: EXIF/IPTC)
+   * 
+   * @param  string $file  Path to the file 
+   * 
+   * @return array  Metadata as array
+   */
+  public function readMetadata(string $file)
+  {
+    return self::readJpegMetadata($file);
+  }
+
+  /**
+   * Reads the EXIF and IPTC metadata from a JPEG.
+   * 
+   * @param  string $file  The image path
+   * 
+   * @return array         Metadata in array format
+   * 
+   * @since 4.0.0
+   */
+  public function readJpegMetadata(string $file)
+  {
+    // Output to the same format as before. Comment field has been left out on purpose.
+    $metadata = array('exif' => array(), 'iptc' => array());
+    $size = getimagesize($file, $info);
+
+    // EXIF with PEL
+    $imageObjects = self::getPelImageObjects($file);
+    if ($imageObjects == false) {
+      return;
+    }
+    $tiff = $imageObjects["tiff"];
+    $ifd0 = $tiff->getIfd();
+    if ($ifd0 != null) {
+      $metadata['exif']['IFD0'] = array();
+      foreach ($ifd0->getEntries() as $entry) {
+        $metadata['exif']['IFD0'][PelTag::getName(PelIfd::IFD0, $entry->getTag())] = self::formatPELEntryForForm($entry);
+      }
+      $subIfd = $ifd0->getSubIfd(PelIfd::EXIF);
+      if ($subIfd != null) {
+        $metadata['exif']['EXIF'] = array();
+        foreach ($subIfd->getEntries() as $entry) {
+          $metadata['exif']['EXIF'][PelTag::getName(PelIfd::EXIF, $entry->getTag())] = self::formatPELEntryForForm($entry);
+        }
+      }
+    }
+
+    // IPTC
+    if (isset($info["APP13"])) {
+      $iptc = iptcparse($info['APP13']);
+      foreach ($iptc as $key => $value) {
+        $metadata['iptc'][$key] = $value[0];
+      }
+    }
+    return $metadata;
+  }
+
+  /**
+   * Gets the jpeg/tiff objects from a valid JPEG or TIFF image with PEL.
+   * 
+   * @param  string $img    The image data
+   * 
+   * @return array|false    File and tiff objects on success, false on failure.
+   * 
+   * @since 4.0.0
+   */
+  private function getPelImageObjects(string $img)
```

Comment:
Why have you chosen them to have the `private` scope and not the `protected` scope?
If you dont have an explicit reason to use `private`, please use `protected`.